docs/sphinx/using/cudaq/platform.rst

```diff
@@ -104,32 +115,90 @@ should be used.
-    :start-after: [Begin Documentation]
+    .. literalinclude:: ../../snippets/python/using/cudaq/platform/observe_mqpu_mpi.py
+        :language: python
+        :start-after: [Begin Documentation]
 
-.. code-block:: console
+    .. code-block:: console
 
-    mpirun -np <N> python3 file.py
+        mpirun -np <N> python3 file.py
 
 In the above example, the parallel distribution mode was set to :code:`mpi` using :code:`cudaq::parallel::mpi` in C++ or :code:`cudaq.parallel.mpi` in Python.
 CUDA Quantum provides MPI utility functions to initialize, finalize, or query (rank, size, etc.) the MPI runtime. 
 Last but not least, the compiled executable (C++) or Python script needs to be launched with an appropriate MPI command, 
 e.g., :code:`mpirun`, :code:`mpiexec`, :code:`srun`, etc.
+
+Remote REST Server Platform
++++++++++++++++++++++++++++
+
+The remote simulator target (:code:`remote-sim`) encapsulates simulated QPUs
+as independent REST server instances, to which the CUDA Quantum runtime communicating 
+to via HTTP requests (REST API).
+
+CUDA Quantum provides the REST server implementation as a standalone application (:code:`cudaq_rest_server`)
+hosting all the simulator backends available in the installation, including those that require MPI for multi-GPU computation.
+To start the server, serving at a specific TCP/IP port, one can do the following.
+
+.. code-block:: console
+    cudaq_rest_server --port <port number>
+
+User code can then target this platform by specifying its target name (:code:`remote-sim`).
+
+.. tab:: C++
+
+    .. code-block:: console
+
+        nvq++ file.cpp --target remote-sim --remote-sim-url <url1[,url2,...]> --remote-sim-backend <sim1[,sim2,...]>
+
+
+.. tab:: Python
+
+     .. code:: python 
+
+        cudaq.set_target("remote-sim", url="url1[,url2,...]", backend="sim1[,sim2,...]")
+    
+
+When using this target, the user needs to provides a list of URLs where (:code:`cudaq_rest_server`) is serving.
+The number of QPUs (:code:`num_qpus()`) is equal to the number of URLs provided. 
+
+Each QPU instance can be assigned a different backend simulator via the :code:`--remote-sim-backend` (`nvq++`) or :code:`backend` (Python)
+option. Otherwise, if a single backend is specified, all the QPUs are assumed to be using the same simulator.
+
+Auto-launch REST Server
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The server app (:code:`cudaq_rest_server`) can be launch and shutdown automatically
+by using the auto-launch feature of the platform.
+Random TCP/IP ports, that are available for use, will be selected to launch those server processes.
+
+.. tab:: C++
+
+    .. code-block:: console
+
+        nvq++ file.cpp --target remote-sim --remote-sim-auto-launch <N> --remote-sim-backend <sim1[,sim2,...]>
+
+
+.. tab:: Python
+
+     .. code:: python 
+
+        cudaq.set_target("remote-sim", auto_launch="<N>", backend="sim1[,sim2,...]")
+
+
+In the above snippets, `N` denotes the number of REST server instances (QPUs) to be launched.
```

Comment:
Q: Does 'N' have an upper limit?

Reply:
It currently has no limit. If unable to find a port, the execution will fail. 
We could certainly cap this `N` param, just not sure what the limit should be.

Reply:
Isn't the max number of ports 65,535? So, one could use up to that as long as free port is available.
I think it is okay to (a) test practical limit, (b) document the known limitations.